File: AircraftSimulatorsControl/DAL/Entity/Component.cs
namespace AircraftSimulatorsControl.DAL.Entity
{
    public class Component : IEntity
    {

        public int ComponentId { get; set; }

        public string Name { get; set; } = "Unknown component";

        public required int SimulatorId { get; set; }
        public required Simulator Simulator { get; set; }

    }

}

File: WebServer/src/main/java/org/server/dal/model/Component.java
package org.server.dal.model;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Table(name = "component")
@Entity
@Data
public class Component implements DatabaseEntity {

    @Id
    @Column(name = "id", unique = true)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name")
    private String name;

    @ManyToOne()
    @JoinColumn(name = "simulator_id")
    private Simulator simulator;

    @OneToMany(mappedBy = "component", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<ComponentStatus> componentStatuses;

    @Override
    public String toString() {
        return "Component{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}
